src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -176,11 +176,28 @@ public static void ProcessLogFile(string filePath, Solution solution, string too
                 log = JsonConvert.DeserializeObject<SarifLog>(logText);
             }
 
-            ProcessSarifLog(log, outputPath, solution);
+            ProcessSarifLog(log, outputPath, solution, showMessageOnNoResults: promptOnLogConversions);
 
             SarifTableDataSource.Instance.BringToFront();
         }
 
+        /// <summary>
+        /// Closes the specified SARIF log in the viewer.
+        /// </summary>
+        /// <param name="paths">The complete path to the SARIF log file.</param>
+        public static void CloseSarifLogs(IEnumerable<string> paths)
+        {
+            SarifTableDataSource.Instance.CleanErrors(paths);
+        }
+
+        /// <summary>
+        /// Closes all SARIF logs opened in the viewer.
+        /// </summary>
+        public static void CloseAllSarifLogs()
+        {
+            SarifTableDataSource.Instance.CleanAllErrors();
+        }
+
         private const string VersionRegexPattern = @"""version""\s*:\s*""(?<version>[\d.]+)""";
```

Comment:
>VersionRegexPattern [](start = 29, length = 19)

This is not cool.  First this should be a Regex type and second it is compiled.  If compiled then it should be cached, but then again this causes a DLL to be created and never unloaded for something that is used what 1 time?  I suggest passing instead

CultureInvariant | ExplicitCapture | Singleline

Reply:
This regex isn't used just once, it's evaluated every time you load a SARIF log file, so IMO it should be compiled once and cached:

```
private const string VersionRegexPattern = @"""version""\s*:\s*""(?<version>[\d.]+)""";
private static readonly s_versionRegex = new Regex(VersionRegexPattern, RegexOptions.Compiled | RegexOptions.CultureInvariant);
```

and then at the point of call:

```
return s_versionRegex.Match(headSegment);
```

`RegexOptions.SingleLine` (treat the input string as a single line by allowing `'.'` to match `'\n'`) isn't necessary because there are no `'.'`s in the pattern (the `'.'` in the character class is a literal `'.'`, not a metacharacter).

`Regex.ExplicitCapture` (don't capture unnamed groups) isn't necessary because there are no unnamed groups in the pattern.

---
In reply to: [440944742](https://github.com/microsoft/sarif-visualstudio-extension/pull/143#discussion_r440944742) [](ancestors = 440944742)